src/OrderFormAcceptanceTests.Actions/Pages/OrderForm.cs

```diff
@@ -316,6 +322,22 @@ public void EnterContact(Contact contact)
 			Driver.FindElement(Pages.OrderForm.ContactTelephone).SendKeys(contact.Phone);
 		}
 
+		public Contact GetContact()
+		{			
+			Wait.Until(d => d.FindElements(Pages.OrderForm.ContactFirstName).Count == 1);
+			var FirstNameElement = Driver.FindElement(Pages.OrderForm.ContactFirstName);
+			var LastNameElement = Driver.FindElement(Pages.OrderForm.ContactLastName);
+			var EmailElement = Driver.FindElement(Pages.OrderForm.ContactEmail);
+			var PhoneElement = Driver.FindElement(Pages.OrderForm.ContactTelephone);
+			return new Contact()
+			{
+				FirstName = FirstNameElement.Text == "" ? FirstNameElement.GetAttribute("value") : FirstNameElement.Text,
+				LastName = LastNameElement.Text == "" ? LastNameElement.GetAttribute("value") : LastNameElement.Text,
+				Email = EmailElement.Text == "" ? EmailElement.GetAttribute("value") : EmailElement.Text,
+				Phone = PhoneElement.Text == "" ? PhoneElement.GetAttribute("value") : PhoneElement.Text,
```

Comment:
Could check to see if the current tagname is `input` rather than checking for the Text of each element first

Reply:
The only reason I'm doing any of it instead of just using the element.Text is because when the info is retrieved it's not accessible as text for some reason. It's part of the element's value instead. So I could just go and always get the value, but then I thought about what if it's accessible as text in another scenario? it might not be, as i suspect it's the way the text value is handled on page load. If the page loads with it, it's an elements value. If the page loads empty and i enter text, it's treated as text.

How would you feel if i just stripped it all out and got the value for now?

Reply:
We can refactor it back if we need to